front/temporal/labs/utils/gong.ts

```diff
@@ -1,9 +1,44 @@
+import { getOAuthConnectionAccessToken } from "@dust-tt/types";
+
+import config from "@app/lib/api/config";
 import type { Authenticator } from "@app/lib/auth";
-import config from "@app/lib/labs/config";
+import labsConfig from "@app/lib/labs/config";
 import { getAccessTokenFromNango } from "@app/lib/labs/transcripts/utils/helpers";
+import { isDualUseOAuthConnectionId } from "@app/lib/oauth";
 import type { LabsTranscriptsConfigurationResource } from "@app/lib/resources/labs_transcripts_resource";
 import type { Logger } from "@app/logger/logger";
 
+const getGongAccessToken = async (
+  transcriptsConfiguration: LabsTranscriptsConfigurationResource,
+  logger: Logger
+) => {
+  if (isDualUseOAuthConnectionId(transcriptsConfiguration.connectionId)) {
+    const tokRes = await getOAuthConnectionAccessToken({
+      config: config.getOAuthAPIConfig(),
+      logger,
+      provider: "gong",
+      connectionId: transcriptsConfiguration.connectionId,
+    });
+    if (tokRes.isErr()) {
+      logger.error(
+        {
+          connectionId: transcriptsConfiguration.connectionId,
+          error: tokRes.error,
+        },
+        "Error retrieving Intercom access token"
+      );
+      throw new Error("Error retrieving Intercom access token");
```

Comment:
@albandum you might want to surface this error differently, with the current implementation the activity will always fail... It's okay for this PR but please make sure to find some time to either move this to connectors or implement proper/more robust error handling.